src/platform/backends/hyperv_api/hyperv_api_operation_result.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#pragma once
+
+#include <winerror.h>
+
+#include <string>
+
+#include <fmt/format.h>
+#include <fmt/std.h>
+
+namespace multipass::hyperv
+{
+
+/**
+ * A simple HRESULT wrapper which is boolable for
+ * convenience.
+ */
+struct ResultCode
+{
+    using unsigned_hresult_t = std::make_unsigned_t<HRESULT>;
+
+    ResultCode(HRESULT r) noexcept : result(r)
```

Comment:
Is this implicit on purpose?

Reply:
Yes:

```c++
    const ResultCode result =
        api.SetComputeSystemCallback(static_cast<HCS_SYSTEM>(target_hcs_system.get()),
                                     HCS_EVENT_OPTIONS::HcsEventOptionNone,
                                     context,
                                     reinterpret_cast<HCS_EVENT_CALLBACK>(callback));
```

... where api.* functions return HRESULT.